Seems the Gradle build is currently broken here:

```
14:08:38 FAILURE: Build failed with an exception.
14:08:38 
14:08:38 * Where:
14:08:38 Build file '/var/lib/jenkins/workspace/elastic+elasticsearch+pull-request-1/plugins/repository-s3/build.gradle' line: 245
14:08:38 
14:08:38 * What went wrong:
14:08:38 A problem occurred evaluating project ':plugins:repository-s3'.
14:08:38 > /var/lib/jenkins/workspace/elastic+elasticsearch+pull-request-1/plugins/repository-s3/build/fixtures/s3Fixture.properties (No such file or directory)
14:08:38 
```

--- a/docs/plugins/repository-s3.asciidoc
+++ b/docs/plugins/repository-s3.asciidoc
@@ -153,6 +153,13 @@ settings belong in the `elasticsearch.yml` file.
    https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Builder.html#setPathStyleAccessEnabled-java.lang.Boolean-[AWS
    documentation] for details). Defaults to `false`.
 
+`disable_chunked_encoding`::
+    Whether chunked encoding should be disabled or not. If `true`, the
+    chunked encoding will be disabled. If `false`, the chunked encoding will
+    be automatically determined by the AWS Java SDK (See
+    https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Builder.html#withChunkedEncodingDisabled-java.lang.Boolean-[AWS
+    documentation] for details). Defaults to `false`.
+
 [[repository-s3-path-style-deprecation]]
 NOTE: In versions `7.0`, `7.1`, `7.2` and `7.3` all bucket operations used the
 https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/[now-deprecated]

--- a/docs/plugins/repository-s3.asciidoc
+++ b/docs/plugins/repository-s3.asciidoc
@@ -153,18 +153,18 @@ settings belong in the `elasticsearch.yml` file.
    https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Builder.html#setPathStyleAccessEnabled-java.lang.Boolean-[AWS
    documentation] for details). Defaults to `false`.
 
+[[repository-s3-path-style-deprecation]]
+NOTE: In versions `7.0`, `7.1`, `7.2` and `7.3` all bucket operations used the
+https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/[now-deprecated]
+path style access pattern. If your deployment requires the path style access
+pattern then you should set this setting to `true` when upgrading.
+
 `disable_chunked_encoding`::
     Whether chunked encoding should be disabled or not. If `true`, the
     chunked encoding will be disabled. If `false`, the chunked encoding will
     be automatically determined by the AWS Java SDK (See
     https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Builder.html#withChunkedEncodingDisabled-java.lang.Boolean-[AWS
     documentation] for details). Defaults to `false`.
-
-[[repository-s3-path-style-deprecation]]
-NOTE: In versions `7.0`, `7.1`, `7.2` and `7.3` all bucket operations used the
-https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/[now-deprecated]
-path style access pattern. If your deployment requires the path style access
-pattern then you should set this setting to `true` when upgrading.
 
 [float]
 [[repository-s3-compatible-services]]

--- a/plugins/repository-s3/build.gradle
+++ b/plugins/repository-s3/build.gradle
@@ -235,7 +235,10 @@ task s3FixtureProperties {
       "s3Fixture.disableChunkedEncoding" : s3DisableChunkedEncoding
   ]
 
-  file(s3FixtureFile).text = s3FixtureOptions.collect { k, v -> "$k = $v" }.join("\n")
+  outputs.upToDateWhen { false }
+  doLast {
+    file(s3FixtureFile).text = s3FixtureOptions.collect { k, v -> "$k = $v" }.join("\n")
+  }
 }
 
 /** A task to start the AmazonS3Fixture which emulates an S3 service **/

--- a/plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java
+++ b/plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java
@@ -99,6 +99,10 @@ final class S3ClientSettings {
     static final Setting.AffixSetting<Boolean> USE_PATH_STYLE_ACCESS = Setting.affixKeySetting(PREFIX, "path_style_access",
         key -> Setting.boolSetting(key, false, Property.NodeScope));
 
+    /** Whether chunked encoding should be disabled or not (Default is false). */
+    static final Setting.AffixSetting<Boolean> DISABLE_CHUNKED_ENCODING = Setting.affixKeySetting(PREFIX, "disable_chunked_encoding",
+        key -> Setting.boolSetting(key, false, Property.NodeScope));
+
     /** Credentials to authenticate with s3. */
     final S3BasicCredentials credentials;
 
@@ -134,10 +138,13 @@ final class S3ClientSettings {
     /** Whether the s3 client should use path style access. */
     final boolean pathStyleAccess;
 
+    /** Whether chunked encoding should be disabled or not. */
+    final boolean disableChunkedEncoding;
+
     private S3ClientSettings(S3BasicCredentials credentials, String endpoint, Protocol protocol,
                              String proxyHost, int proxyPort, String proxyUsername, String proxyPassword,
                              int readTimeoutMillis, int maxRetries, boolean throttleRetries,
-                             boolean pathStyleAccess) {
+                             boolean pathStyleAccess, boolean disableChunkedEncoding) {
         this.credentials = credentials;
         this.endpoint = endpoint;
         this.protocol = protocol;
@@ -149,6 +156,7 @@ final class S3ClientSettings {
         this.maxRetries = maxRetries;
         this.throttleRetries = throttleRetries;
         this.pathStyleAccess = pathStyleAccess;
+        this.disableChunkedEncoding = disableChunkedEncoding;
     }
 
     /**
@@ -172,6 +180,8 @@ final class S3ClientSettings {
         final int newMaxRetries = getRepoSettingOrDefault(MAX_RETRIES_SETTING, normalizedSettings, maxRetries);
         final boolean newThrottleRetries = getRepoSettingOrDefault(USE_THROTTLE_RETRIES_SETTING, normalizedSettings, throttleRetries);
         final boolean usePathStyleAccess = getRepoSettingOrDefault(USE_PATH_STYLE_ACCESS, normalizedSettings, pathStyleAccess);
+        final boolean newDisableChunkedEncoding = getRepoSettingOrDefault(
+            DISABLE_CHUNKED_ENCODING, normalizedSettings, disableChunkedEncoding);
         final S3BasicCredentials newCredentials;
         if (checkDeprecatedCredentials(repoSettings)) {
             newCredentials = loadDeprecatedCredentials(repoSettings);
@@ -180,7 +190,8 @@ final class S3ClientSettings {
         }
         if (Objects.equals(endpoint, newEndpoint) && protocol == newProtocol && Objects.equals(proxyHost, newProxyHost)
             && proxyPort == newProxyPort && newReadTimeoutMillis == readTimeoutMillis && maxRetries == newMaxRetries
-            && newThrottleRetries == throttleRetries && Objects.equals(credentials, newCredentials)) {
+            && newThrottleRetries == throttleRetries && Objects.equals(credentials, newCredentials)
+            && newDisableChunkedEncoding == disableChunkedEncoding) {
             return this;
         }
         return new S3ClientSettings(
@@ -194,7 +205,8 @@ final class S3ClientSettings {
             newReadTimeoutMillis,
             newMaxRetries,
             newThrottleRetries,
-            usePathStyleAccess
+            usePathStyleAccess,
+            newDisableChunkedEncoding
         );
     }
 
@@ -282,7 +294,8 @@ final class S3ClientSettings {
                 Math.toIntExact(getConfigValue(settings, clientName, READ_TIMEOUT_SETTING).millis()),
                 getConfigValue(settings, clientName, MAX_RETRIES_SETTING),
                 getConfigValue(settings, clientName, USE_THROTTLE_RETRIES_SETTING),
-                getConfigValue(settings, clientName, USE_PATH_STYLE_ACCESS)
+                getConfigValue(settings, clientName, USE_PATH_STYLE_ACCESS),
+                getConfigValue(settings, clientName, DISABLE_CHUNKED_ENCODING)
             );
         }
     }
@@ -305,13 +318,14 @@ final class S3ClientSettings {
             protocol == that.protocol &&
             Objects.equals(proxyHost, that.proxyHost) &&
             Objects.equals(proxyUsername, that.proxyUsername) &&
-            Objects.equals(proxyPassword, that.proxyPassword);
+            Objects.equals(proxyPassword, that.proxyPassword) &&
+            Objects.equals(disableChunkedEncoding, that.disableChunkedEncoding);
     }
 
     @Override
     public int hashCode() {
         return Objects.hash(credentials, endpoint, protocol, proxyHost, proxyPort, proxyUsername, proxyPassword,
-            readTimeoutMillis, maxRetries, throttleRetries);
+            readTimeoutMillis, maxRetries, throttleRetries, disableChunkedEncoding);
     }
 
     private static <T> T getConfigValue(Settings settings, String clientName,

--- a/plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java
+++ b/plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java
@@ -157,6 +157,10 @@ class S3Service implements Closeable {
         if (clientSettings.pathStyleAccess) {
             builder.enablePathStyleAccess();
         }
+
+        if (clientSettings.disableChunkedEncoding) {
+            builder.withChunkedEncodingDisabled(true);
+        }
         return builder.build();
     }
 

--- a/plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java
+++ b/plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java
@@ -157,9 +157,8 @@ class S3Service implements Closeable {
         if (clientSettings.pathStyleAccess) {
             builder.enablePathStyleAccess();
         }
-
         if (clientSettings.disableChunkedEncoding) {
-            builder.withChunkedEncodingDisabled(true);
+            builder.disableChunkedEncoding();
         }
         return builder.build();
     }

--- a/plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java
+++ b/plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java
@@ -151,4 +151,11 @@ public class S3ClientSettingsTests extends ESTestCase {
         assertThat(settings.get("default").pathStyleAccess, is(false));
         assertThat(settings.get("other").pathStyleAccess, is(true));
     }
+
+    public void testUseChunkedEncodingCanBeSet() {
+        final Map<String, S3ClientSettings> settings = S3ClientSettings.load(
+            Settings.builder().put("s3.client.other.disable_chunked_encoding", true).build());
+        assertThat(settings.get("default").disableChunkedEncoding, is(false));
+        assertThat(settings.get("other").disableChunkedEncoding, is(true));
+    }
 }

